### PR TITLE
Update run_test script to run the API integration tests in the new standalone environment

### LINKED ISSUE #10594: Update run_test.py to pass AIT with the new `standalone` and `cluster` marks
|Component|
|---|
| API Integration Tests |

This issue is related to the epic issue #10137. In that epic, new manager standalone API integration tests are told to be implemented.

After the addition of the new marks: `standalone` and `cluster`, the API Integration tests are going to be passed with the following commands:

```
pytest -m standalone test_x.tavern.yaml
pytest -m cluster test_x.tavern.yaml
```

With the first command, all tests with the `standalone` mark will be passed in a environment without a cluster deployed.
With the second command, all tests with the `cluster` mark will be passed in a Wazuh cluster environment.

In this issue, we should adapt our `run_test.py` script to execute both commands. With both executions, some tests will be passed in both environments, others will be passed in standalone mode, and the rest, in only cluster mode. We should also take into account that we should merge both results. Also, we need to investigate how to return the test results, as some of the tests are going to be passed twice, as I said.

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

### PR DESCRIPTION
**Pull request by mcarmona99**
|Related issue|
|---|
|#10594|

This PR closes #10594.

The run_test.py script has been updated to pass AIT using the new `standalone` and `cluster` marks.

The usage is the same, there aren't any new parameters or different behaviours. 

Some of the functions or code we already had in the script have been improved. Docstrings have also been added to make the code more human-readable.

`run_tests.py` examples:

### `-l` parameter

```
$ python3 run_tests.py -l test_cdb_list,test_overview
Collected tests [2]:
test_cdb_list_endpoints.tavern.yaml, test_overview_endpoints.tavern.yaml


test_cdb_list_endpoints.tavern.yaml - Standalone environment 
	 1 passed, 4 deselected, 9 warnings

test_cdb_list_endpoints.tavern.yaml - Cluster environment 
	 2 passed, 3 deselected, 9 warnings

test_overview_endpoints.tavern.yaml - Standalone environment 
	 1 passed, 4 warnings

test_overview_endpoints.tavern.yaml - Cluster environment 
	 1 deselected, 4 warnings
```

### `-k` and `-i`

```
$ python3 run_tests.py -k test_cdb -i 2
Collected tests [1]:
test_cdb_list_endpoints.tavern.yaml


test_cdb_list_endpoints.tavern.yaml - Standalone environment [1/2]
	 1 passed, 4 deselected, 9 warnings

test_cdb_list_endpoints.tavern.yaml - Cluster environment [1/2]
	 2 passed, 3 deselected, 9 warnings

test_cdb_list_endpoints.tavern.yaml - Standalone environment [2/2]
	 1 passed, 4 deselected, 9 warnings

test_cdb_list_endpoints.tavern.yaml - Cluster environment [2/2]
	 2 passed, 3 deselected, 9 warnings
```

```
$ ls _test_results/
 html_reports   logs  'test_cdb_list_endpoints_cluster'  'test_cdb_list_endpoints_cluster2'  'test_cdb_list_endpoints_standalone'  'test_cdb_list_endpoints_standalone2'
```

```
$ ls _test_results/html_reports/
'test_cdb_list_endpoints_cluster.html'  'test_cdb_list_endpoints_cluster2.html'  'test_cdb_list_endpoints_standalone.html'  'test_cdb_list_endpoints_standalone2.html'
```

### `-e` parameter

I only added the standalone and cluster marks to overview and cdb list endpoints API integration tests. This is why all are deselected.

```
$ python3 run_tests.py -e
Collected tests [64]:
test_active_response_endpoints.tavern.yaml, test_agent_DELETE_endpoints.tavern.yaml, test_agent_GET_endpoints.tavern.yaml, test_agent_POST_endpoints.tavern.yaml, test_agent_PUT_endpoints.tavern.yaml, test_ciscat_endpoints.tavern.yaml, test_cluster_endpoints.tavern.yaml, test_decoder_endpoints.tavern.yaml, test_default_endpoints.tavern.yaml, test_experimental_endpoints.tavern.yaml, test_logtest_endpoints.tavern.yaml, test_manager_endpoints.tavern.yaml, test_mitre_endpoints.tavern.yaml, test_overview_endpoints.tavern.yaml, test_rbac_black_active_response_endpoints.tavern.yaml, test_rbac_black_agent_endpoints.tavern.yaml, test_rbac_black_cdb_list_endpoints.tavern.yaml, test_rbac_black_ciscat_endpoints.tavern.yaml, test_rbac_black_cluster_endpoints.tavern.yaml, test_rbac_black_decoder_endpoints.tavern.yaml, test_rbac_black_experimental_endpoints.tavern.yaml, test_rbac_black_logtest_endpoints.tavern.yaml, test_rbac_black_manager_endpoints.tavern.yaml, test_rbac_black_mitre_endpoints.tavern.yaml, test_rbac_black_overview_endpoints.tavern.yaml, test_rbac_black_rootcheck_endpoints.tavern.yaml, test_rbac_black_rule_endpoints.tavern.yaml, test_rbac_black_sca_endpoints.tavern.yaml, test_rbac_black_security_endpoints.tavern.yaml, test_rbac_black_syscheck_endpoints.tavern.yaml, test_rbac_black_syscollector_endpoints.tavern.yaml, test_rbac_black_task_endpoints.tavern.yaml, test_rbac_black_vulnerability_endpoints.tavern.yaml, test_rbac_white_active_response_endpoints.tavern.yaml, test_rbac_white_agent_endpoints.tavern.yaml, test_rbac_white_all_endpoints.tavern.yaml, test_rbac_white_cdb_list_endpoints.tavern.yaml, test_rbac_white_ciscat_endpoints.tavern.yaml, test_rbac_white_cluster_endpoints.tavern.yaml, test_rbac_white_decoder_endpoints.tavern.yaml, test_rbac_white_experimental_endpoints.tavern.yaml, test_rbac_white_logtest_endpoints.tavern.yaml, test_rbac_white_manager_endpoints.tavern.yaml, test_rbac_white_mitre_endpoints.tavern.yaml, test_rbac_white_overview_endpoints.tavern.yaml, test_rbac_white_rootcheck_endpoints.tavern.yaml, test_rbac_white_rule_endpoints.tavern.yaml, test_rbac_white_sca_endpoints.tavern.yaml, test_rbac_white_security_endpoints.tavern.yaml, test_rbac_white_syscheck_endpoints.tavern.yaml, test_rbac_white_syscollector_endpoints.tavern.yaml, test_rbac_white_task_endpoints.tavern.yaml, test_rbac_white_vulnerability_endpoints.tavern.yaml, test_rootcheck_endpoints.tavern.yaml, test_rule_endpoints.tavern.yaml, test_sca_endpoints.tavern.yaml, test_security_DELETE_endpoints.tavern.yaml, test_security_GET_endpoints.tavern.yaml, test_security_POST_endpoints.tavern.yaml, test_security_PUT_endpoints.tavern.yaml, test_syscheck_endpoints.tavern.yaml, test_syscollector_endpoints.tavern.yaml, test_task_endpoints.tavern.yaml, test_vulnerability_endpoints.tavern.yaml


test_active_response_endpoints.tavern.yaml - Standalone environment 
	 2 deselected, 4 warnings

test_active_response_endpoints.tavern.yaml - Cluster environment 
	 2 deselected, 4 warnings

test_agent_DELETE_endpoints.tavern.yaml - Standalone environment 
	 6 deselected, 8 warnings

test_agent_DELETE_endpoints.tavern.yaml - Cluster environment 
	 6 deselected, 8 warnings

...

test_mitre_endpoints.tavern.yaml - Standalone environment 
	 7 deselected, 13 warnings

test_mitre_endpoints.tavern.yaml - Cluster environment 
	 7 deselected, 13 warnings

test_overview_endpoints.tavern.yaml - Standalone environment 
	 1 passed, 4 warnings

test_overview_endpoints.tavern.yaml - Cluster environment 
	 1 deselected, 4 warnings

...

test_vulnerability_endpoints.tavern.yaml - Standalone environment 
	 1 deselected, 3 warnings

test_vulnerability_endpoints.tavern.yaml - Cluster environment 
	 1 deselected, 3 warnings
```

### `-r` parameter

```
$ python3 run_tests.py -r
test_cdb_list_endpoints_standalone- Standalone environment
	 1 passed, 4 deselected, 9 warnings

test_cdb_list_endpoints_standalone2- Standalone environment
	 1 passed, 4 deselected, 9 warnings

test_cdb_list_endpoints_cluster- Cluster environment
	 2 passed, 3 deselected, 9 warnings

test_cdb_list_endpoints_cluster2- Cluster environment
	 2 passed, 3 deselected, 9 warnings
```

### `-rbac` parameter

```
$ python3 run_tests.py -k cdb -rbac yes
Collected tests [2]:
test_rbac_black_cdb_list_endpoints.tavern.yaml, test_rbac_white_cdb_list_endpoints.tavern.yaml


test_rbac_black_cdb_list_endpoints.tavern.yaml 
	 5 passed, 7 warnings

test_rbac_white_cdb_list_endpoints.tavern.yaml 
	 5 passed, 7 warnings
```

```
$ python3 run_tests.py -k cdb -rbac no
Collected tests [1]:
test_cdb_list_endpoints.tavern.yaml


test_cdb_list_endpoints.tavern.yaml - Standalone environment 
	 1 passed, 4 deselected, 9 warnings

test_cdb_list_endpoints.tavern.yaml - Cluster environment 
	 2 passed, 3 deselected, 9 warnings
```

```
$ python3 run_tests.py -k cdb -rbac both
Collected tests [3]:
test_cdb_list_endpoints.tavern.yaml, test_rbac_black_cdb_list_endpoints.tavern.yaml, test_rbac_white_cdb_list_endpoints.tavern.yaml


test_cdb_list_endpoints.tavern.yaml - Standalone environment 
	 1 passed, 4 deselected, 9 warnings

test_cdb_list_endpoints.tavern.yaml - Cluster environment 
	 2 passed, 3 deselected, 9 warnings

test_rbac_black_cdb_list_endpoints.tavern.yaml 
	 5 passed, 7 warnings

test_rbac_white_cdb_list_endpoints.tavern.yaml 
	 5 passed, 7 warnings
```

### CODE REVIEW

**Comment by mcarmona99** [comment]
## PR update

A new parameter has been added to the script. This parameter is `-mode` and it is used to choose between passing the tests in a standalone mode, cluster mode, or both modes (default).

The parameter is similar to the `-rbac` parameter.

```
$ python3 run_tests.py -k test_cdb -mode standalone
Collected tests [1]:
test_cdb_list_endpoints.tavern.yaml


test_cdb_list_endpoints.tavern.yaml - Standalone environment 
	 1 failed, 4 passed, 9 warnings


$ python3 run_tests.py -k test_cdb -mode standalon
usage: run_tests.py [options]
run_tests.py: error: argument -mode: invalid choice: 'standalon' (choose from 'both', 'standalone', 'cluster')


$ python3 run_tests.py -k test_cdb -mode cluster
Collected tests [1]:
test_cdb_list_endpoints.tavern.yaml


test_cdb_list_endpoints.tavern.yaml - Cluster environment 
...


$ python3 run_tests.py -k test_cdb -mode both
Collected tests [1]:
test_cdb_list_endpoints.tavern.yaml


test_cdb_list_endpoints.tavern.yaml - Standalone environment 
	 1 failed, 4 passed, 9 warnings

test_cdb_list_endpoints.tavern.yaml - Cluster environment 
	 1 failed, 4 passed, 9 warnings
```

File: api/test/integration/run_tests.py [hunk 1]
```diff
@@ -20,105 +32,214 @@ def calculate_result(file_name):
...
     group.add_argument('-l', '--list', dest='test_list', default=None,
                        help='Specify a list of tests separated by a comma.', action='store')
     group.add_argument('-e', '--exclude', dest='exclude', action='store_true', default=None,
-                       help='Run every test excluding the already saved in the RESULTS_PATH.')
+                       help='Run every test excluding the already saved in the RESULTS_FOLDER.')
     group.add_argument('-r', '--results', dest='results', action='store_true', default=None,
                        help='Get result summary from the already run tests.')
     parser.add_argument('-k', '--keyword', dest='keyword', default=None,
                         help='Specify the keyword to filter tests out. Default None.', action='store')
     parser.add_argument('-rbac', dest='rbac', default='both', choices=rbac_choices,
```
Comment (vicferpoy) [inline api/test/integration/run_tests.py:217]:
```suggestion
    parser.add_argument('-b', '--rbac', dest='rbac', default='both', choices=rbac_choices,
```

File: api/test/integration/run_tests.py [hunk 2]
```diff
@@ -20,105 +32,214 @@ def calculate_result(file_name):
...
                         help='Specify what to do with RBAC tests. Run everything, only RBAC ones or no RBAC. Default '
                              '"both".', action='store')
+    parser.add_argument('-mode', dest='mode', default='both', choices=mode_choices,
```
Comment (vicferpoy) [inline api/test/integration/run_tests.py:220]:
```suggestion
    parser.add_argument('-m', '--mode', dest='mode', default='both', choices=mode_choices,
```

**Review by vicferpoy — CHANGES_REQUESTED** [review]

File: api/test/integration/run_tests.py (hunk 1 above)
Comment (mcarmona99) [reply api/test/integration/run_tests.py:217]: Done

File: api/test/integration/run_tests.py (hunk 2 above)
Comment (mcarmona99) [reply api/test/integration/run_tests.py:220]: Done

**Comment by mcarmona99** [comment]
## PR update

Script output with the new flags:

```
$ python3 run_tests.py -k cdb --mode=cluster -b no
Collected tests [1]:
test_cdb_list_endpoints.tavern.yaml


test_cdb_list_endpoints.tavern.yaml - Cluster environment 
	 1 failed, 4 deselected, 9 warnings
```

```
$ python3 run_tests.py -k overview --rbac=no
Collected tests [1]:
test_overview_endpoints.tavern.yaml


test_overview_endpoints.tavern.yaml - Standalone environment 
	 1 passed, 5 warnings

test_overview_endpoints.tavern.yaml - Cluster environment 
	 1 passed, 5 warnings
```

```
$ python3 run_tests.py -k overview -m standalone
Collected tests [3]:
test_overview_endpoints.tavern.yaml, test_rbac_black_overview_endpoints.tavern.yaml, test_rbac_white_overview_endpoints.tavern.yaml


test_overview_endpoints.tavern.yaml - Standalone environment 
	 1 passed, 5 warnings

test_rbac_black_overview_endpoints.tavern.yaml 
	 1 passed, 3 warnings

test_rbac_white_overview_endpoints.tavern.yaml 
	 1 passed, 3 warnings
```

**Review by vicferpoy — APPROVED** [review]
LGTM

File: api/test/integration/run_tests.py [hunk 3]
```diff
@@ -20,105 +32,214 @@ def calculate_result(file_name):
...
                         help='Specify how many times will every test be run. Default 1.', action='store')
 
     return parser.parse_args()
 
 
 if __name__ == '__main__':
-    os.makedirs(os.path.join(TESTS_PATH, RESULTS_PATH), exist_ok=True)
-    os.makedirs(os.path.join(TESTS_PATH, RESULTS_PATH, 'html_reports'), exist_ok=True)
+    os.makedirs(os.path.join(TESTS_PATH, RESULTS_FOLDER), exist_ok=True)
+    os.makedirs(os.path.join(TESTS_PATH, RESULTS_FOLDER, 'html_reports'), exist_ok=True)
```
Comment (vicferpoy) [inline api/test/integration/run_tests.py:231]:
```suggestion
    os.makedirs(os.path.join(TESTS_PATH, RESULTS_FOLDER, 'html_reports'), exist_ok=True)
```

File: api/test/integration/run_tests.py [hunk 4]
```diff
@@ -20,105 +32,214 @@ def calculate_result(file_name):
...
                        help='Specify a list of tests separated by a comma.', action='store')
     group.add_argument('-e', '--exclude', dest='exclude', action='store_true', default=None,
-                       help='Run every test excluding the already saved in the RESULTS_PATH.')
+                       help='Run every test excluding the already saved in the RESULTS_FOLDER.')
     group.add_argument('-r', '--results', dest='results', action='store_true', default=None,
                        help='Get result summary from the already run tests.')
     parser.add_argument('-k', '--keyword', dest='keyword', default=None,
                         help='Specify the keyword to filter tests out. Default None.', action='store')
-    parser.add_argument('-rbac', dest='rbac', default='both', choices=rbac_choices,
+    parser.add_argument('-b', '--rbac', dest='rbac', default='both', choices=rbac_choices,
```
Comment (vicferpoy) [inline api/test/integration/run_tests.py:217]:
```suggestion
    parser.add_argument('-R', '--rbac', dest='rbac', default='both', choices=rbac_choices,
```

**Review by vicferpoy — CHANGES_REQUESTED** [review]

**Review by vicferpoy — APPROVED** [review]
LGTM

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**